cpp/include/cudf/detail/utilities/hash_functions.cuh

```diff
@@ -66,7 +65,12 @@ void CUDA_DEVICE_CALLABLE md5_hash_step(md5_intermediate_data* hash_state,
         break;
     }
 
-    F = F + A + hash_constants[j] + buffer_ints[g];
+    uint32_t buffer_element_as_int;
+    thrust::copy_n(thrust::seq,
+                   hash_state->buffer + g * 4,
+                   4,
+                   reinterpret_cast<uint8_t*>(&buffer_element_as_int));
```

Comment:
```suggestion
   uint32_t& buffer_element_as_int = reinterpret_cast<uint32_t*>(hash_state->buffer)[g];
```

Reply:
This is what existed before, and it's undefined behavior because it violates the strict aliasing rules. See https://gist.github.com/shafik/848ae25ee209f698763cffee272a58f8